Effects set in home assistant have no effect

I'm using v1.10.0-dev.8, FUT089 and home assistant and it seems there's an issue on triggering effects from home assistant interface.

This is my relevant hass config:
```
- platform: mqtt
  name: 'Living room downlights 1'
  qos: 1
  schema: json
  color_temp: true
  rgb: true
  brightness: true
  retain: true
  command_topic: milight/0x0/fut089/1
  state_topic: milight/states/0x0/fut089/1
  availability_topic: milight/client_status
  payload_available: 'connected'
  payload_not_available: 'disconnected'
  effect: true
  #Effect list 0=Full, 1=White, 2=RGB, 3=Full hard, 4=Disco, 5=Red Flash, 6=Green Flash, 7=Blue Flash, 8=White Flash]
  effect_list: [0, 1, 2, 3, 4, 5, 6, 7, 8]
```

When I trigger effect 2 in hass, the following mqtt messages are sent and nothing happens:
```
milight/0x0/fut089/1 {"state": "ON", "effect": "2"}
milight/updates/0x0/fut089/1 {"state":"ON"}
milight/states/0x0/fut089/1 {"state":"ON","brightness":125,"color_temp":370,"bulb_mode":"white","effect":"white_mode"}
```

However, if I trigger mode 2 in ESP8266 web interface, the following mqtt messages are sent:
```
milight/updates/0x0/fut089/1 {"mode":2}
milight/states/0x0/fut089/1 {"state":"ON","mode":2,"bulb_mode":"scene","effect":"2"}
```

So, it seems to me that in order for the effect to work, besides "effect": "2", home assistant should also send "bulb_mode": "scene". 
I'm not sure if home assistant is expected to send the "bulb_mode" attribute or the firmware should automatically set "bulb_mode": "scene" whenever "effect" != "white_mode".

Whenever I'm changing from white to color (in home assistant), "bulb_mode" is adjusted accordingly so I have a feeling this is really a bug on milight_hub software.

For some reason 1.9.2 doesn't build for me, do you have any idea?
```
.pio/libdeps/nodemcuv2/WebSockets_ID549/src/WebSocketsClient.cpp: In member function 'void WebSocketsClient::connectedCb()':
.pio/libdeps/nodemcuv2/WebSockets_ID549/src/WebSocketsClient.cpp:749:22: error: 'class axTLS::WiFiClientSecure' has no member n
amed 'setInsecure'
         _client.ssl->setInsecure();

```

--- a/lib/MiLight/MiLightClient.cpp
+++ b/lib/MiLight/MiLightClient.cpp
@@ -20,6 +20,7 @@ const char* MiLightClient::FIELD_ORDERINGS[] = {
   GroupStateFieldNames::TEMPERATURE,
   GroupStateFieldNames::COLOR_TEMP,
   GroupStateFieldNames::MODE,
+  GroupStateFieldNames::EFFECT,
   GroupStateFieldNames::COLOR,
   // Level/Brightness must be processed last because they're specific to a particular bulb mode.
   // So make sure bulb mode is set before applying level/brightness.

--- a/test/remote/settings.json.example
+++ b/test/remote/settings.json.example
@@ -30,6 +30,7 @@
     "status",
     "level",
     "color_temp",
+    "kelvin",
     "bulb_mode",
     "hue",
     "saturation",

--- a/test/remote/spec/state_spec.rb
+++ b/test/remote/spec/state_spec.rb
@@ -374,6 +374,14 @@ RSpec.describe 'State' do
       # Should retain previous brightness
       expect(result['level']).to eq(50)
     end
+
+    it 'should support the mode and effect fields' do
+      state = @client.patch_state({status: 'ON', mode: 0}, @id_params)
+      expect(state['effect']).to eq("0")
+
+      state = @client.patch_state({effect: 1}, @id_params)
+      expect(state['effect']).to eq("1")
+    end
   end
 
   context 'increment/decrement commands' do

